src/map/lua/lua_baseentity.cpp

```diff
@@ -4402,7 +4402,11 @@ inline int32 CLuaBaseEntity::canUsePet(lua_State *L)
 {
     DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
 
-    lua_pushboolean(L, m_PBaseEntity->loc.zone->CanUseMisc(MISC_PET));
+    if (m_PBaseEntity->loc.zone->CanUseMisc(MISC_PET) && ((CCharEntity*)m_PBaseEntity)->m_moghouseID == 0)
+        lua_pushboolean(L, true);
+    else
+        lua_pushboolean(L, false);
+
```

Comment:
you'll want to check if the entity is actually a player (in case mobs can call this too) so `dynamic_cast<CCharEntity*>(m_PBaseEntity)`
if the cast doesnt return nullptr you know its a player, so can check `PChar->loc.zone->CanUseMisc(MISC_PET) && PChar->m_moghouseID == 0` otherwise only need to check the zone's misc pet flag